client/src/components/Timer.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect, useState } from "react";
+import { useGame } from "@/context/GameContext";
+
+export default function Timer() {
+    
+    const { hours, minutes, seconds} = {hours: 0, minutes: 0, seconds: 0};
+    const [[hrs, mins, secs], setTime] = useState([hours, minutes, seconds]);
+    const game = useGame();
+
+    const tick = () => {
+        if (mins === 59 && secs === 59) {
+            setTime([hrs + 1, 0, 0]);
+        } else if (mins !== 59 && secs === 59) {
+            setTime([hrs, mins + 1, 0]);
+        } else {
+            setTime([hrs, mins, secs + 1]);
+        }
+    };
+
+    function reset() {
+        setTime([0, 0, 0]);
+    }
+
+    useEffect(() => {
+        reset();
+    }, [game.level]);
+
+    useEffect(() => {
+        const timerId = setInterval(() => tick(), 1000);
+        return () => clearInterval(timerId);
+    });
+
+    return (
+        <div>
+            {`${hrs.toString().padStart(2, '0')}:${mins
+            .toString()
+            .padStart(2, '0')}:${secs.toString().padStart(2, '0')}`}
+        </div>
```

Comment:
nit: Consider wrapping this in a `<span />` tag instead of `<div />`. This will increase flexibility as future consumers of this component may not always want it to be in a `div` (which can affect layout).

Reply:
I would suggest actually replacing the `div` with a chakra-ui `<Box />` component and forwarding any props to the `Box` component. This way, anyone that needs to style the layout of the timer can directly pass valid properties as defined by `BoxProps`. Something like:

```jsx
const Timer:React.FC<BoxProps> = ({ ...rest }) => {
   // stuff
   <Box {...rest} >
     {time}
   </Box>
}
```

Reply:
Consider using `font-family: monospace` for the timer. This will use a font type that makes each character equal space which prevents the timer values from shrinking and expanding depending on the values. i.e. a "1" will take up the same horizontal space as "0".

Reply:
These changes are implemented in the recent commit

Reply:
I don't think `BoxProps` was implemented. You have to `import { BoxProps } from "@chakra-ui/react"`;` and then can set the `TimerProps` as `export type TimerProps = {} & BoxProps;`. Then just do what @garethlau showed above.

Reply:
Ok I implemented the above!